cypress/support/commands/commands.js

```diff
@@ -203,6 +203,70 @@ Cypress.Commands.add(
     }
 );
 
+/**
+ * Types in a sw-multi-select field all the specified values and checks if the content was correctly set.
+ * @memberOf Cypress.Chainable#
+ * @name typeMultiSelectAndCheckMultiple
+ * @function
+ * @param {String[]} values - Desired values of the element
+ */
+Cypress.Commands.add(
+    "typeMultiSelectAndCheckMultiple",
+    {
+        prevSubject: "element",
+    },
+    (subject, values) => {
+        // Request we want to wait for later
+        cy.server();
+        cy.route({
+            url: `${Cypress.env("apiPath")}/search/*`,
+            method: "post",
+        }).as("filteredResultCall");
+
+        cy.wrap(subject).should("be.visible");
+
+        // type in the search term if available
+        for (let i = 0; i < values.length; i += 1) {
+            cy.get(`${subject.selector} .sw-select-selection-list__input`)
+                .clear()
+                .type(values[i])
+                .should(
+                    "have.value",
+                    values[i]
+                );
+
+            // wait for the first request (which happens on opening / clicking in the input
+            cy.wait("@filteredResultCall").then(() => {
+                // waitt for the second request (which happens on stop typing with the actual search)
+                cy.wait("@filteredResultCall").then(() => {
+                    cy.get(".sw-loader__element").should("not.exist");
+                });
+            });
+
+            // select the value
+            cy.contains(`.sw-select-result-list__content .sw-select-result`, values[i]).should('be.visible').click();
+        }
+
+        // close search results
+        cy.get(`${subject.selector} .sw-select-selection-list__input`).type("{esc}");
+        cy.get(`${subject.selector} .sw-select-result-list`).should(
+            "not.exist"
+        );
+
+        // check if all values are selected
+        let selectedValues = cy.get(
```

Comment:
Storing a return value is a [Cypress anti-pattern](https://docs.cypress.io/guides/references/best-practices.html#Assigning-Return-Values). Please try to avoid that.

Reply:
Okay, I didn't know that (I am still a cypress beginner :smile: ). So I should call `cy.get(...)` in a loop even if it requests the same element over and over again (is it cached internally)?

Reply:
I recommend using closures inside the loop to react accordingly - see: https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html#Closures

Reply:
Changed the loop to multiple `cy.get(...)` calls to prevent this anti-pattern.